src/controls/MyTeams/MyTeams.tsx

```diff
@@ -0,0 +1,147 @@
+  props: React.PropsWithChildren<IMyTeamsProps>
+) => {
+  const {
+    stackStyles,
+    stackTokens,
+    styleClasses,
+    titleStyles,
+  } = getMyTeamsStyles(props.themeVariant);
+
+  const { title, webPartContext, onSelectedChannel, themeVariant, enablePersonCardInteraction } = props;
+  const [state, dispatch] = React.useReducer(myTeamsReducer, initialState);
+
+  Providers.globalProvider = React.useMemo(() => {
+    return new SharePointProvider(webPartContext);
+  }, [props.webPartContext]);
+
+  const { getMyTeams } = useTeams(webPartContext.serviceScope);
+
+  React.useEffect(() => {
+    (async () => {
+      try {
+        dispatch({
+          type: EMyTeamsTypes.SET_HAS_ERROR,
+          payload: false
+        });
+        dispatch({
+          type:  EMyTeamsTypes.SET_IS_LOADING,
+          payload: true
+        });
+        const _teams: ITeam[] = await getMyTeams();
+        dispatch({
+          type: EMyTeamsTypes.SET_IS_LOADING,
+          payload: false
+        });
+        dispatch({
+          type: EMyTeamsTypes.SET_MESSAGE,
+          payload: undefined
+        });
+        dispatch({
+          type: EMyTeamsTypes.SET_MYTEAMS,
+          payload: _teams
+        });
+      } catch (error) {
+        const messageError: IShowMessageProps = {
+          isShow: true,
+          message: strings.MyTeamsMessageError,
+          messageBarType: MessageBarType.error,
+        };
+        dispatch({
+          type: EMyTeamsTypes.SET_IS_LOADING,
+          payload: false
+        });
+        dispatch({
+          type: EMyTeamsTypes.SET_MESSAGE,
+          payload: messageError
+        });
+        dispatch({
+          type: EMyTeamsTypes.SET_HAS_ERROR,
+          payload: true
+        });
+      }
+    })();
+  }, []);
+
+
+  const {hasError, isLoading, message, myTeams } = state;
+
+  return (
+    <>
+      <Customizer settings={{ theme:  themeVariant }}>
+        <Stack styles={stackStyles} tokens={stackTokens}>
+          <Text variant={"xLargePlus"} styles={titleStyles}>
+            {title}
+          </Text>
+          {hasError ? (
+            <ShowMessage {...message} />
+          ) : (
+            <div className={styleClasses.teamsContainer}>
+              {!myTeams?.length && !isLoading ? (
+                <Stack
+                  horizontal
+                  horizontalAlign="center"
+                  verticalAlign="center"
+                  tokens={{ childrenGap: 20 }}
+                  styles={{ root: { margin: 20 } }}
+                >
+                  <FontIcon
+                    iconName="TeamsLogo"
+                    style={{ fontSize: 28, width: 28, height: 28 }}
+                  />
+                  <Text variant="mediumPlus">
+                    {strings.MyTeamsNoTeamsMessage}
+                  </Text>
+                </Stack>
+              ) : isLoading ? (
+                <Spinner
+                  size={SpinnerSize.medium}
+                  label="loading you teams"
```

Comment:
@joaojmendes can you please make this a loc string so that it can be translated?

Reply:
created string